libs/sr-opt/src/parentWorker.ts

```diff
@@ -0,0 +1,177 @@
+import type { NumTagFree } from '@genshin-optimizer/pando'
+import { combineConst, flatten } from '@genshin-optimizer/pando'
+import {
+  allRelicSlotKeys,
+  type RelicSlotKey,
+} from '@genshin-optimizer/sr-consts'
+import type { ICachedRelic } from '@genshin-optimizer/sr-db'
+import { getRelicMainStatVal } from '@genshin-optimizer/sr-util'
+import { objKeyMap, range } from '@genshin-optimizer/util'
+import type { ChildCommand, ChildMessage } from './childWorker'
+import type { BuildResult } from './optimize'
+
+export interface ParentCommandStart {
+  command: 'start'
+  relicsBySlot: Record<RelicSlotKey, ICachedRelic[]>
+  detachedNodes: NumTagFree[]
+  numWorkers: number
+  maxResults: number
+}
+export type ParentCommand = ParentCommandStart
+
+export interface ParentMessageProgress {
+  resultType: 'progress'
+  numBuilds: number
+  numBuildsComputed: number
+  sendingResults?: boolean
+}
+export interface ParentMessageSending {
+  resultType: 'sending'
+}
+export interface ParentMessageDone {
+  resultType: 'done'
+  buildResults: BuildResult[]
+}
+export interface ParentMessageErr {
+  resultType: 'err'
+  message: string
+}
+export type ParentMessage =
+  | ParentMessageProgress
+  | ParentMessageDone
+  | ParentMessageErr
+
+export type RelicStats = {
+  id: string
+  stats: Record<string, number>
+}
+
+// Get proper typings for posting a message back to main thread
+declare function postMessage(message: ParentMessage): void
+
+// Receiving a message from main thread to worker
+onmessage = async (e: MessageEvent<ParentCommand>) => {
+  try {
+    await handleEvent(e)
+  } catch (err) {
+    console.log(err)
+    postMessage({ resultType: 'err', message: JSON.stringify(err) })
+  }
+}
+
+async function handleEvent(e: MessageEvent<ParentCommand>): Promise<void> {
+  const { data } = e,
+    { command } = data
+  switch (command) {
+    case 'start':
+      await start(data)
+  }
+}
+
+async function start({
+  relicsBySlot,
+  detachedNodes,
+  numWorkers,
+  maxResults,
+}: ParentCommandStart) {
+  // Step 3: Optimize nodes, as needed
+  const flatNodes = flatten(detachedNodes)
+  const combinedNodes = combineConst(flatNodes)
```

Comment:
It might be better to keep reusing the name `detachedNodes`, i.e. `detachedNodes = combineConst(...)`. The result is supposed to be "the same computation but faster". If this comes from the example code, maybe change that too.